core/capabilities/testutils/chain_reader.go

```diff
@@ -0,0 +1,178 @@
+
+	reqConfig.ContractReaderConfig = contractReaderCfgMap
+
+	config, err := commonvalues.WrapMap(reqConfig)
+	require.NoError(t, err)
+	req := commoncaps.TriggerRegistrationRequest{
+		TriggerID: "logeventtrigger_log1",
+		Config:    config,
+		Metadata: commoncaps.RequestMetadata{
+			ReferenceID: "logeventtrigger",
+		},
+	}
+
+	// Create a new contract reader to return from mock relayer
+	ctx := coretestutils.Context(t)
+	contractReader, err := backendTH.NewContractReader(ctx, t, contractReaderCfgBytes)
+	require.NoError(t, err)
+
+	return &ContractReaderTH{
+		BackendTH: backendTH,
+
+		LogEmitterAddress:           &logEmitterAddress,
+		LogEmitterContract:          logEmitter,
+		LogEmitterContractReader:    contractReader,
+		LogEmitterRegRequest:        req,
+		LogEmitterContractReaderCfg: contractReaderCfgBytes,
+	}
+}
+
+// Wait for a specific log to be emitted to a response channel by ChainReader
+func WaitForLog(lggr logger.Logger, logCh <-chan commoncaps.TriggerResponse, timeout time.Duration) (
+	*commoncaps.TriggerResponse, map[string]any, error) {
+	select {
+	case <-time.After(timeout):
+		return nil, nil, fmt.Errorf("timeout waiting for Log1 event from ContractReader")
+	case log := <-logCh:
+		lggr.Infow("Received log from ContractReader", "event", log.Event.ID)
+		if log.Err != nil {
+			return nil, nil, fmt.Errorf("error listening for Log1 event from ContractReader: %v", log.Err)
+		}
+		v := make(map[string]any)
+		err := log.Event.Outputs.UnwrapTo(&v)
+		if err != nil {
+			return nil, nil, fmt.Errorf("error unwrapping log to map: (log %v) %v", log.Event.Outputs, log.Err)
+		}
+		return &log, v, nil
+	}
+}
+
+// Get the string value of a key from a generic map[string]any
+func GetStrVal(m map[string]any, k string) (string, error) {
+	v, ok := m[k]
+	if !ok {
+		return "", fmt.Errorf("key %s not found", k)
+	}
+	vstr, ok := v.(string)
+	if !ok {
+		return "", fmt.Errorf("key %s not a string (%T)", k, v)
+	}
+	return vstr, nil
+}
+
+// Get int value of a key from a generic map[string]any
+func GetBigIntVal(m map[string]any, k string) (*big.Int, error) {
+	v, ok := m[k]
+	if !ok {
+		return nil, fmt.Errorf("key %s not found", k)
+	}
+	val, ok := v.(*big.Int)
+	if !ok {
+		return nil, fmt.Errorf("key %s not a *big.Int (%T)", k, v)
+	}
+	return val, nil
+}
+
+// Get the int value from a map[string]map[string]any
+func GetBigIntValL2(m map[string]any, level1Key string, level2Key string) (*big.Int, error) {
+	v, ok := m[level1Key]
+	if !ok {
+		return nil, fmt.Errorf("key %s not found", level1Key)
+	}
+	level2Map, ok := v.(map[string]any)
+	if !ok {
+		return nil, fmt.Errorf("key %s not a map[string]any (%T)", level1Key, v)
+	}
+	return GetBigIntVal(level2Map, level2Key)
+}
+
+// Pretty print a map as a JSON string
+func PrintMap(m map[string]any, prefix string, lggr logger.Logger) error {
+	json, err := json.MarshalIndent(m, "", "  ")
+	if err != nil {
+		return err
+	}
+	lggr.Infow(prefix, "map", string(json))
```

Comment:
These kinds of logger helpers don't work well because the caller information is lost. How about returning the `string(json)` to log from the call site?

Reply:
This is intended just for tests

Reply:
It is still important to follow standards in tests.

Reply:
Removed this method and logged at the callsite